src/main/java/teammates/ui/controller/InstructorStudentListPageData.java

```diff
@@ -1,22 +1,86 @@
 package teammates.ui.controller;
 
-import java.util.HashMap;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
 import teammates.common.datatransfer.AccountAttributes;
 import teammates.common.datatransfer.CourseAttributes;
 import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+import teammates.ui.template.InstructorStudentsListFilterBox;
+import teammates.ui.template.InstructorStudentsListFilterCourses;
+import teammates.ui.template.InstructorStudentsListSearchBox;
+import teammates.ui.template.InstructorStudentsListStudentsTableCourses;
 
 public class InstructorStudentListPageData extends PageData {
 
-    public InstructorStudentListPageData(AccountAttributes account) {
+    private InstructorStudentsListSearchBox searchBox;
+    private InstructorStudentsListFilterBox filterBox;
+    private List<InstructorStudentsListStudentsTableCourses> studentsTable;
+    private int numOfCourses;
+    public List<CourseAttributes> courses;
+    public Boolean displayArchive;
+    public String searchKey;
+    
+    public InstructorStudentListPageData(AccountAttributes account, String searchKey,
+                                         boolean displayArchive,
+                                         Map<String, InstructorAttributes> instructors,
+                                         Map<String, String> numStudents,
+                                         List<CourseAttributes> courses) {
         super(account);
+        this.courses = courses;
+        this.displayArchive = displayArchive;
+        this.searchKey = searchKey;
+        init(account, searchKey, displayArchive, instructors, numStudents, courses);
     }
+    
+    // data.isCourseArchived, course.id, data.account.googleId, data.numStudents.get(course.id),
+    // data.getInstructorCourseEnrollLink(course.id), course.name
```

Comment:
x__x forgot to delete.
(side note: this is how I decide which fields are necessary to be included in the PageData models)